packages/babel-parser/src/plugins/estree.js

```diff
@@ -411,4 +411,28 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         super.toAssignableObjectExpressionProp(prop, isBinding, isLast);
       }
     }
+
+    finishCallExpression<T: N.CallExpression | N.OptionalCallExpression>(
+      node: T,
+      optional: boolean,
+    ): N.Expression {
+      super.finishCallExpression(node, optional);
+
+      if (node.callee.type === "Import") {
+        ((node: N.Node): N.EstreeImportExpression).type = "ImportExpression";
+        ((node: N.Node): N.EstreeImportExpression).source = node.arguments[0];
+        delete ((node: N.Node): N.EstreeImportExpression).arguments;
+        delete ((node: N.Node): N.EstreeImportExpression).callee;
+      }
+
+      return node;
+    }
+
+    // ImportExpressions do not have an arguments array.
+    toReferencedListDeep(
+      exprList: $ReadOnlyArray<?N.Expression> = [],
```

Comment:
This was crashing because it's given the `CallExpression`'s `arguments` array, which is `undefined` for `ImportExpression`s (deleted above).

Reply:
What about doing this? We avoid using the empty array and iterating it:

```js
toReferencedListDeep(exprList: ?$ReadOnlyArray<?N.Expression>) {
  if (exprList) return super.toReferencedListDeep(exprList);
}
```

If flow complains that it can't return undefined, we can just change the original `toReferencedListDeep` signature to return `void`, since it's return value is never used.

Reply:
Sure, that sounds good to me. Updated!